src/game.cpp

```diff
@@ -11483,7 +11483,15 @@ void game::reload( int pos )
     }
 
     auto loc = it->pick_reload_ammo( u, true );
-    if( loc.get_item() ) {
+
+    const auto ammo = loc.get_item();
+    if( ammo ) {
+        if( ammo->is_magazine() && ammo->ammo_remaining() == 0 ) {
+            if( !query_yn( _( "Reload using an empty magazine?" ) ) ) {
```

Comment:
It would be better if the the menu handled it instead.
Alternating menus and y/n queries is annoying and they are confirmed in different ways (enter confirms menu, but cancels query).

As in, if you are trying to reload with an empty magazine, show a menu as if multiple magazines existed.

Always having a menu (even for single full magazine) would also be good. You could then have reload button confirm the reload, like it is the case with vehicle repairs and butchery.